### DISCUSSION TITLE
How do you trap focus in a dialog?

### DISCUSSION THREAD

**Discussion by candidia** [ariakit/ariakit#2615] [Questions] [answered]
Sorry I think this is probably simple, but can't quite get it. For [accessibility](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/) ("modal dialogs do not provide means for moving keyboard focus outside the dialog window without closing the dialog") I would like to loop/trap focus in my dialog. I'm using the `modal` prop but tabbing at the final tabbable shifts focus outside my app whereas I want it to go back to the first tabbable in the dialog. I don't see a prop that looks like it would loop focus. Am I missing it?

**Comment by diegohaz** [comment, marked as answer]
That's working as intended. And that's how the native modal [`<dialog>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog#result_2) works as well.

Keyboard and screen reader users should be able to access the same elements as mouse users, who can always interact with elements outside your site frame when a web modal dialog is open. When you trap focus when the user presses <kbd>Tab</kbd>, you restrict access for keyboard users. More details on [Focus trap elements](https://ariakit.org/examples/dialog-radix#focus-trap-elements).

**Reply by candidia** [reply]
Ok thanks I think I was misled on this. Seems like it is a common misconception https://github.com/w3c/html/issues/1514

**Reply by diegohaz** [reply]
Yeah, most component libraries (including Ariakit in the past) do the focus trap thing. And the way it's implemented in the WAI-ARIA pattern also contributes to that.
